R/metab.bookkeep.R

```diff
@@ -73,7 +73,7 @@ metab.bookkeep <- function(do.obs, do.sat, k.gas, z.mix, irr, ...){
 		warning("datetime not found, inferring sampling frequency from # of observations") # issue a warning (note checks in addNAs)
 		# NOTE: because of the checks in addNA's, it is unlikely a user would receive this warning via metab()
 		# warning will only be seen through direct use of metab.bookkeep when datettime is not supplied
-		freq <- nobs
+		freq <- nobs - 1
```

Comment:
12AM*, but yeah, I know what you mean. And of course, this would be the
intuitive thing to do if you had a bunch of observations over many days ---
for any diff() value, you'd assign the time stamp to the time of
origination not termination; so do(11PM) - do(12AM) would be a delta-do
value assign to 11PM's day (because the process occurs then).

As far as what we have to change, this will affect metab() in a big way I'm
pretty sure. This is because you can no longer split the data set by the
calendar day and do all calculations on each piece. Honestly, this is
probably why I didn't include 2 midnights to begin with: it was just more
natural to code it with 1 midnight.

On Wed, Sep 7, 2016 at 3:03 PM, Luke Winslow <notifications@github.com>
wrote:

> In R/metab.bookkeep.R
> <https://github.com/GLEON/LakeMetabolizer/pull/121#discussion_r77883998>:
>
> > @@ -73,7 +73,7 @@ metab.bookkeep <- function(do.obs, do.sat, k.gas, z.mix, irr, ...){
> >  		warning("datetime not found, inferring sampling frequency from # of observations") # issue a warning (note checks in addNAs)
> >  		# NOTE: because of the checks in addNA's, it is unlikely a user would receive this warning via metab()
> >  		# warning will only be seen through direct use of metab.bookkeep when datettime is not supplied
> > -		freq <- nobs
> > +		freq <- nobs - 1
>
> If you stop at 11PM, you are missing all the metabolism and gas exchange
> occurring on day X between 11 and 12PM. All process is occurring on day X,
> you are just using the midnight value as a boundary condition for the day.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/GLEON/LakeMetabolizer/pull/121/files/c75567a84f7677ee3238146a630d9ff014c790da#r77883998>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFXrfwcTT0M1EmXHbGZwnT4HkgghyHxUks5qnwptgaJpZM4Jvljp>
> .
>



-- 
Ryan D. Batt
Aquatic Ecologist
Postdoctoral Researcher, Rutgers University
Website: http://limnology.wisc.edu/personnel/batt/